flink-end-to-end-tests/flink-end-to-end-tests-common/src/main/java/org/apache/flink/tests/util/flink/FlinkDistribution.java

```diff
@@ -223,6 +221,22 @@ public void submitSQLJob(SQLJobSubmission job, Duration timeout) throws IOExcept
                 .runBlocking(timeout);
     }
 
+    public void performJarAddition(JarAddition addition) throws IOException {
+        final Path target = mapJarLocationToPath(addition.getTarget());
+        final Path sourceJar = addition.getJar();
+
+        final String jarNameWithoutExtension =
+                FilenameUtils.removeExtension(sourceJar.getFileName().toString());
```

Comment:
Why do we remove the file extension here? 🤔

Reply:
because creating directories with file extensions is a bit weird isn't it

Reply:
oh, I misinterpreted the `resolve` method chain. 🤦 makes sense